modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -0,0 +1,207 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#include "test_precomp.hpp"
+
+
+#include "logger.hpp"
+#include "common/gapi_tests_common.hpp"
+#include "opencv2/gapi/gpu/ggpukernel.hpp"
+#include "opencl_kernels_test_gapi.hpp"
+
+
+namespace cv
+{
+
+#ifdef HAVE_OPENCL
+
+    static void reference_symm7x7_CPU(const cv::Mat& in, const cv::Mat& kernel_coeff, int shift, cv::Mat &out)
+    {
+        cv::Point anchor = { -1, -1 };
+        double delta = 0;
+
+        const int* ci = kernel_coeff.ptr<int>();
+
+        float c_float[10];
+        float divisor = (float)(1 << shift);
+        for (int i = 0; i < 10; i++)
+        {
+            c_float[i] = ci[i] / divisor;
+        }
+        // J & I & H & G & H & I & J
+        // I & F & E & D & E & F & I
+        // H & E & C & B & C & E & H
+        // G & D & B & A & B & D & G
+        // H & E & C & B & C & E & H
+        // I & F & E & D & E & F & I
+        // J & I & H & G & H & I & J
+
+        // A & B & C & D & E & F & G & H & I & J
+
+        // 9 & 8 & 7 & 6 & 7 & 8 & 9
+        // 8 & 5 & 4 & 3 & 4 & 5 & 8
+        // 7 & 4 & 2 & 1 & 2 & 4 & 7
+        // 6 & 3 & 1 & 0 & 1 & 3 & 6
+        // 7 & 4 & 2 & 1 & 2 & 4 & 7
+        // 8 & 5 & 4 & 3 & 4 & 5 & 8
+        // 9 & 8 & 7 & 6 & 7 & 8 & 9
+
+        float coefficients[49] =
+        {
+            c_float[9], c_float[8], c_float[7], c_float[6], c_float[7], c_float[8], c_float[9],
+            c_float[8], c_float[5], c_float[4], c_float[3], c_float[4], c_float[5], c_float[8],
+            c_float[7], c_float[4], c_float[2], c_float[1], c_float[2], c_float[4], c_float[7],
+            c_float[6], c_float[3], c_float[1], c_float[0], c_float[1], c_float[3], c_float[6],
+            c_float[7], c_float[4], c_float[2], c_float[1], c_float[2], c_float[4], c_float[7],
+            c_float[8], c_float[5], c_float[4], c_float[3], c_float[4], c_float[5], c_float[8],
+            c_float[9], c_float[8], c_float[7], c_float[6], c_float[7], c_float[8], c_float[9]
+        };
+
+        cv::Mat kernel = cv::Mat(7, 7, CV_32FC1);
```

Comment:
`cv::Mat` can be initialized directly from that array, given `stride = 7*sizeof(float)`